[Bug 760] Fix setting WEP key in monitor mode causes IV lost.

Signed-off-by: Zhu Yi <yi.zhu@intel.com>
--
diff -urp ipw2200-1.0.6/ipw2200.c ipw2200-1.0.6-monitor_wep/ipw2200.c
--- ipw2200-1.0.6/ipw2200.c	2005-08-25 11:13:57.000000000 +0800
+++ ipw2200-1.0.6-monitor_wep/ipw2200.c	2005-08-25 11:14:19.000000000 +0800
@@ -7648,7 +7648,8 @@ static void ipw_handle_data_packet(struc
 	IPW_DEBUG_RX("Rx packet of %d bytes.\n", rxb->skb->len);
 
 	/* HW decrypt will not clear the WEP bit, MIC, PN, etc. */
-	if (!priv->ieee->host_decrypt)
+	if (!priv->ieee->host_decrypt &&
+	    priv->ieee->iw_mode != IW_MODE_MONITOR)
 		ipw_rebuild_decrypted_skb(priv, rxb->skb);
 
 	if (!ieee80211_rx(priv->ieee, rxb->skb, stats))
